spec/paper_trail/cleaner_spec.rb

```diff
@@ -58,7 +58,7 @@ module PaperTrail
           expect(animal.paper_trail.versions_between(date, (date + 1.day)).size).to(eq(3))
           PaperTrail.clean_versions!(date: date)
           expect(PaperTrail::Version.count).to(eq(8))
-          expect(animal.versions.reload.size).to(eq(2))
+          expect(animal.versions.size).to(eq(2))
```

Comment:
Why remove this `reload`?

I'd expect a memory optimization to *not* change existing tests.

Reply:
@jaredbeck The existing tests are able to pass without any changes. I've removed that because `reload` call is rudimental now:

1. We going to change the way how to build the Version objects, in order to optimize the memory usage. Now we use pure `Class.new` instead of `association.build`;
2. That will fail some tests since `Class.new` has no side effects around association cache;
3. in order to fix that we invalidate the association cache implicitly by `association.reset` right after any Version object saving;
4. at this point the `versions.reload` becomes to be rudimental. I.e. `versions` association works as expected even after the PT callbacks without any additional efforts.

I can move `reloads` back if you like. In additional, I can add some tests which will prove that `reload` calls are no more necessary :)